crates/ev-cli/src/commands/interact.rs

```diff
@@ -1,39 +1,29 @@
-use dialoguer::Input;
-
 use crate::theme::CliTheme;
+use dialoguer::{Input, Select};
+use indicatif::{ProgressBar, ProgressStyle};
 
 pub mod validators {
     use lazy_static;
     use regex::Regex;
-    use std::fmt::{Debug, Display};
+    use std::fmt::Debug;
+    use thiserror::Error;
 
-    #[derive(Debug)]
+    #[derive(Debug, Error)]
     pub enum ValidationError {
+        #[error("Value must be a valid HTTPS domain")]
         InvalidHostname,
+        #[error("Invalid custom domain. You should not use a top-level domain name, or include a protocol or path")]
         InvalidCustomDomain,
+        #[error("Destination Domain must be a valid hostname and not contain a protocol or path.")]
         InvalidDestinationDomain,
-    }
-    impl Display for ValidationError {
-        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-            #[allow(unreachable_patterns)]
-            match self {
-                Self::InvalidHostname => write!(f, "Value must be a valid HTTPS domain"),
-                Self::InvalidCustomDomain => write!(
-                    f,
-                    "Invalid custom domain. You should not use a \
-                 top-level domain name, or include a protocol or path"
-                ),
-                Self::InvalidDestinationDomain => write!(
-                    f,
-                    "Destination Domain must be a valid hostname and not contain a protocol or path."
-                ),
-                _ => write!(f, "{:?}", self),
-            }
-        }
+        #[error("Invalid function name. Must be between 2 and 40 characters, and contain only alphanumeric characters, dashes, and underscores")]
+        InvalidFunctionName,
+        #[error("Invalid function language. Must be one of: node@18, node@20, python@3.9, python@3.10, python@3.11")]
```

Comment:
todo: will add a section on this to docs and add the link here (also feels like this comment could drift easily, will make a note)

Reply:
Yeah, I'm not sure we should be putting the language versions in the error message. Feels like it's destined to drift

Reply:
Could be easier to update the validation to check that the language is (node|python)@, and then let the API handle deprecation & messaging

Reply:
yea I like that